cfme/tests/integration/test_openldap_auth.py

```diff
@@ -12,13 +12,21 @@ def setup_first_provider():
     setup_a_provider(validate=True, check_existing=True)
 
 
-def test_openldap_auth(request, setup_first_provider, configure_openldap_auth_mode):
+def pytest_generate_tests(metafunc):
     data = cfme_data.get("openldap_test", {})
     if not data:
         pytest.skip("No openldap_test section in yaml")
     group = Group(description=data["group_name"], role="EvmRole-user")
-    request.addfinalizer(group.delete)
-    group.create()
+    metafunc.parametrize(
+        "group_create", [getattr(group, 'create'), getattr(group, 'add_group_from_ldap_lookup')],
+        ids=['create', 'add_group_from_ldap_lookup'])
+
+
+def test_lookup_openldap_groups(
+        request, group_create, setup_first_provider, configure_openldap_auth_mode):
+    data = cfme_data.get("openldap_test", {})
+    group_create()
+    request.addfinalizer(group_create.im_class().delete)
```

Comment:
`im_class().delete` will initialize a new instance of the class with no parameters and then pass its `delete` method. What you want actually is `im_self.delete` in such case.